operators/src/opencl/cl_program.rs

```diff
@@ -1532,6 +1749,271 @@ __kernel void points(
         assert_eq!(collection.multipoint_offsets(), &[0, 1, 3, 4]);
     }
 
+    #[test]
+    fn lines_to_points() {
+        let input = TypedFeatureCollection::MultiLineString(
+            MultiLineStringCollection::from_data(
+                vec![
+                    MultiLineString::new(vec![
+                        vec![(0.0, 0.1).into(), (1.0, 1.1).into()],
+                        vec![(2.0, 2.1).into(), (3.0, 3.1).into(), (4.0, 4.1).into()],
+                    ])
+                    .unwrap(),
+                    MultiLineString::new(vec![vec![(5.0, 5.1).into(), (6.0, 6.1).into()]]).unwrap(),
+                ],
+                vec![
+                    TimeInterval::new_unchecked(0, 1),
+                    TimeInterval::new_unchecked(1, 2),
+                ],
+                HashMap::new(),
+            )
+            .unwrap(),
+        );
+
+        let mut out = FeatureCollection::<MultiPoint>::builder().batch_builder(2, 7);
+
+        let kernel = r#"
+            __kernel void lines_to_points( 
+                    __constant const double2 *IN_FEATURE0_COORDS,
+                    const int IN_FEATURE0_LEN,
+                    __constant const int *IN_FEATURE0_OFFSETS,
+                    __constant const int *IN_FEATURE0_LINE_OFFSETS,
+                    __global double2 *OUT_FEATURE0_COORDS,
+                    __global int *OUT_FEATURE0_OFFSETS
+            ) {
+                int idx = get_global_id(0);
+                
+                OUT_FEATURE0_COORDS[idx].x = IN_FEATURE0_COORDS[idx].x;
+                OUT_FEATURE0_COORDS[idx].y = IN_FEATURE0_COORDS[idx].y + 0.1;
+                
+                if (idx <= IN_FEATURE0_LEN) {
```

Comment:
need to check output len too

Reply:
in the case that the output length is wrong, then this check won't let the program output something valid, so is it still necessary?

Reply:
If we check that we only read valid data from input arrays we should only write data to valid positions in the output array. Of course you can argue that its the caller responsibility to allocate enough memory but I'm not so sure. Additionaly maybe there are algorithms which are flexible in the output length, e.g. a histogram opencl kernel might need the information about the number of buckets which is equal to the length of the output array. Also it would be more uniform if all arrays come as a duo of pointer and length.

Reply:
e172d81

Okay, I see the point where the out length determines the algorithm. However, OpenCL is an interplay between runtime parameter specification and kernel work, so I don't see checking the output lengths in this kernel is helpful.